### PR TITLE
Updated the release-notes doc with some minor updates about internal clean up and a dependency version update.

### PR DESCRIPTION
**Pull request by wdschei**

### CODE REVIEW

File: repose-aggregator/src/docs/asciidoc/release-notes.adoc [hunk 1]
```diff
@@ -3,10 +3,13 @@
 include::_includes/in-progress.adoc[]
 
 == Future Release
+* https://repose.atlassian.net/browse/REP-4465[REP-4465] - Internal clean up converting some enums to constants.
```
Comment (adrianjgeorge) [inline repose-aggregator/src/docs/asciidoc/release-notes.adoc]: This belongs on master not here, yes we'll have to deal with the merge, but i'd rather do that then have someone accidentally drop it into the release notes for a version on this branch.

File: repose-aggregator/src/docs/asciidoc/release-notes.adoc [hunk 2]
```diff
@@ -3,10 +3,13 @@
...
+* https://repose.atlassian.net/browse/REP-4970[REP-4970] - Internal clean up removing an unnecessary dependency.
 
 == 8.4.0.2
 
-* https://repose.atlassian.net/browse/REP-5001[REP-5001] - <<filters/rate-limiting.adoc#, Rate Limiting Filter>> was mistakenly getting the full parameter map, and not just the query parameters.
+* https://repose.atlassian.net/browse/REP-5100[REP-5100] - <<filters/rate-limiting.adoc#, Rate Limiting Filter>> was mistakenly getting the full parameter map, and not just the query parameters.
```
Comment (adrianjgeorge) [inline repose-aggregator/src/docs/asciidoc/release-notes.adoc]: Why does it think there is a change here?

File: repose-aggregator/src/docs/asciidoc/release-notes.adoc [hunk 3]
```diff
@@ -23,4 +26,4 @@ include::_includes/in-progress.adoc[]
 
 == 8.3.0.1 (2016-12-13)
 
-* https://repose.atlassian.net/browse/REP-4764[REP-4764] - sendError in the response wrapper will now call sendError on the underlying response when appropriate.
\ No newline at end of file
+* https://repose.atlassian.net/browse/REP-4764[REP-4764] - sendError in the response wrapper will now call sendError on the underlying response when appropriate.
```
Comment (adrianjgeorge) [inline repose-aggregator/src/docs/asciidoc/release-notes.adoc]: And here? I'm just not seeing them.

File: repose-aggregator/src/docs/asciidoc/release-notes.adoc (hunk 2 above)
Comment (dmnjohns) [inline repose-aggregator/src/docs/asciidoc/release-notes.adoc]: `REP-5001` -> `REP-5100`

File: repose-aggregator/src/docs/asciidoc/release-notes.adoc (hunk 3 above)
Comment (dmnjohns) [inline repose-aggregator/src/docs/asciidoc/release-notes.adoc]: EOF newline, I'm guessing?

File: repose-aggregator/src/docs/asciidoc/release-notes.adoc (hunk 2 above)
Comment (wdschei) [inline repose-aggregator/src/docs/asciidoc/release-notes.adoc]: That was the typo.

File: repose-aggregator/src/docs/asciidoc/release-notes.adoc (hunk 3 above)
Comment (wdschei) [inline repose-aggregator/src/docs/asciidoc/release-notes.adoc]: Yep. But weird it isn't showing the normal red circle thing on the other one.

**Pull request merged**